### PR TITLE
improve license distribution and clarity

### LINKED ISSUE #2958: Make license obligations easier and clearer to understand
* [x] ~~~The wording of LICENSE.txt should be improved to make it unambiguous that normal, expected use does not constitute source redistribution.~~~

* [x] LICENSE.txt file should be included within the installed python lib of installed`conda` packages and `sdist` tarballs. 

* [ ] The text of the LICENSE.txt should be embedded directly into BokehJS. 

* [x] Suggest a `bokeh.license()` function to print the text of the license in the console. 

* [x] When moving from `grunt` to `gulp` the inclusion of license headers in minified BokehJS was lost. This must be restored and maintained under test (suggest grep for one of the known headers as sufficient)

### PR DESCRIPTION
**Pull request by bryevdv**
issues: closes #2958

### CODE REVIEW

**Comment by bryevdv** [comment]
I can't say I love moving the `LICENSE.txt` from the top level into `bokeh` but a quick search revealed only terrible hacks to keep it there, and also distribute it. I'd rather just move it and spend time working on actual things.

**Comment by mattpap** [comment]
@bryevdv, I think the license should be left at the top-level. I didn't yet see a project that would ship it the way it's proposed here. However, if we proceed like this, make sure to update the README and other documentation.

**Comment by bryevdv** [comment]
@mattpap I'm happy to entertain other solutions to keep it at the top level, but I didn't find any that were good. `package_data` requires the extra files to be in the package (and so does `MANIFEST.in` apparently, we have not been shipping the `LICENSE.txt at all in packages even though we thought we were) and `data_files` does not provide a way to easily install into the package (just relpaths from `sys.prefix`).

**Comment by bryevdv** [comment]
We could also jut duplicate it, it's not going to change.

**Comment by bryevdv** [comment]
Or maybe `setup.py` can copy the file to `bokeh` and we can also add a `.gitignore`, thoughts welcome.

File: bokeh/tests/test_bokeh_init.py [hunk 1]
```diff
@@ -16,5 +18,9 @@ def test_version_defined(self):
         import bokeh
         self.assertTrue(bokeh.__version__ != 'unknown')
 
+    def test_license(self):
+        import bokeh
+        self.assertTrue(exists(join(bokeh.__path__[0], 'LICENSE.txt')))
```
Comment (almarklein) [inline bokeh/tests/test_bokeh_init.py]: Would it not be easier/better to test that `license()` does not raise an error?

Comment (bryevdv) [inline bokeh/tests/test_bokeh_init.py]: Yes, thanks!

**Comment by bryevdv** [comment]
@mattpap I have changed it to copy from the top level into the package before running `setup`, seems to work fine. Ping @damianavila for review as well. As soon as @mistakevin has a commit that concats the `LICENSE.txt` text into BokehJS as a comment, I think we can consider this done.

**Comment by bryevdv** [comment]
failing flake8:

https://travis-ci.org/bokeh/bokeh/jobs/84335109#L2611

Should pass when thats fixed up

**Comment by damianavila** [comment]
LGTM

>As soon as @mistakevin has a commit that concats the LICENSE.txt text into BokehJS as a comment,

I guess this needs to be added...

Btw, I restarted the build hanging for 17 hours... (TravisCI had problem yesterday).

**Comment by mistakevin** [comment]
@damianavila @bryevdv @mattpap Ready for review

**Comment by damianavila** [comment]
See TravisCI failure... happy times...

**Comment by mistakevin** [comment]
Weird. It's passing for me locally. Any insight on how to debug this? @damianavila @bryevdv

**Comment by bryevdv** [comment]
That's weird it looks like it failed a string comparison assertion in just py2.7. Are there any stray non-ASCII characters getting thrown in somehow? I can't think of what else might cause it to pass 3.x but fail 2.x tests.

**Comment by bryevdv** [comment]
Restarting the test, just in case this was some spurious issue.

**Comment by mistakevin** [comment]
I'm swamped right now, and can't manage to dedicate any more time to debugging this at the moment. It seems like we want this in ASAP, correct?

**Comment by bryevdv** [comment]
pretty sure it was the fancy quotes.

**Comment by damianavila** [comment]
> pretty sure it was the fancy quotes.

Yep...

**Comment by bryevdv** [comment]
Will merge when this passes (assuming it will)

**Pull request merged**